packages/twenty-server/src/engine/metadata-modules/serverless-function/code-introspection/code-introspection.service.ts

```diff
@@ -0,0 +1,90 @@
+import { Injectable } from '@nestjs/common';
+
+import {
+  ArrowFunction,
+  FunctionDeclaration,
+  Project,
+  SyntaxKind,
+} from 'ts-morph';
+
+import {
+  CodeIntrospectionException,
+  CodeIntrospectionExceptionCode,
+} from 'src/engine/metadata-modules/serverless-function/code-introspection/code-introspection.exception';
+
+type FunctionParameter = {
+  name: string;
+  type: string;
+};
+
+@Injectable()
+export class CodeIntrospectionService {
+  private project: Project;
+
+  constructor() {
+    this.project = new Project();
+  }
+
+  public analyze(
```

Comment:
would be great to have some testing, to see quickly how this analyse function works

Reply:
done